apps/omg_api/test/state/core_test.exs

```diff
@@ -56,24 +60,24 @@ defmodule OMG.API.State.CoreTest do
     state
     |> Test.do_deposit(alice, %{amount: 10, currency: eth(), blknum: 1})
     |> (&Core.exec(
-          Test.create_recovered([{1, 0, 0, alice}], not_eth(), [{bob, 7}, {alice, 3}]),
+          Test.create_recovered([alice], [{1, 0, 0}], [{bob, not_eth(), 7}, {alice, not_eth(), 3}]),
           zero_fees_map(),
           &1
         )).()
-    |> fail?(:incorrect_currency)
+    |> fail?(:input_and_output_currencies_do_not_match)
   end
 
   @tag fixtures: [:alice, :bob, :state_empty]
-  test "when spending inputs must have the same currency", %{alice: alice, bob: bob, state_empty: state} do
+  test "input currencies do not match output currencies", %{alice: alice, bob: bob, state_empty: state} do
     state
     |> Test.do_deposit(alice, %{amount: 10, currency: eth(), blknum: 1})
     |> Test.do_deposit(alice, %{amount: 0, currency: not_eth(), blknum: 2})
     |> (&Core.exec(
-          Test.create_recovered([{1, 0, 0, alice}, {2, 0, 0, alice}], eth(), [{bob, 7}, {alice, 3}]),
+          Test.create_recovered([alice], [{1, 0, 0}, {2, 0, 0}], [{bob, eth(), 7}, {alice, eth(), 3}]),
```

Comment:
please, let's try reverting this change in `create_recovered` API and then reapproach this review.

I'd even opt for having a backwards-compatibility version with `create_recovered/signed(inputs_with_owners, single_currency, outputs)`, just for single-currency transactions, to prevent this storm of changes here.

skipping the review of these test for now